src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,174 @@
+import seedu.duke.recipe.Step;
+import seedu.duke.recipe.StepList;
+import seedu.duke.ui.StringLib;
+import seedu.duke.ui.UI;
+
+import java.util.ArrayList;
+
+import static seedu.duke.ui.IntLib.RECIPE_NAME_INDEX;
+import static seedu.duke.ui.IntLib.RECIPE_INGREDIENTS_INDEX;
+import static seedu.duke.ui.IntLib.RECIPE_TAG_INDEX;
+import static seedu.duke.ui.IntLib.RECIPE_SUM_OF_STEPS_INDEX;
+
+public class Parser {
+
+    private static final String RECIPE_WRONG_NAME_INGREDIENTS_TAG_STEP = "Recipe is missing the \"NAME\" "
+            + "or \"INGREDIENTS\" or \"TAG\" or \"SUM of the STEPs" +
+            "\n or there is more than one \"NAME\" or \"INGREDIENTS\" or \"TAG\" or \"SUM of the STEPs\"!\n";
+    private static final String RECIPE_WRONG_LEADING_STRING = "Recipe contains the leading string!\n";
+    private static final String RECIPE_MISSING_NAME = "Recipe is missing \"NAME\"!\n";
+    private static final String RECIPE_MISSING_INGREDIENTS = "Recipe is missing \"INGREDIENTS\"!\n";
+    private static final String RECIPE_MISSING_TAG = "Recipe is missing \"TAG\"!\n";
+    private static final String RECIPE_MISSING_STEP = "Recipe is missing \"SUM of the STEPs\"!\n";
+    /**
+     * Returns a <code>Command</code> type which contains the command to be
+     * executed and its full description to support its execution. It takes
+     * in a string representing the user input.
+     *
+     * @param line the input line by user.
+     * @return a <code>Command</code> containing the command to be executed.
+     */
+    public static Command parseCommands(String line) {
+        String[] lineSpaced = line.split(" ");
+        String fullDescription = "";
+        if (lineSpaced.length > 1) {
+            for (int i = 2; i < lineSpaced.length; i++) {
+                lineSpaced[1] = lineSpaced[1].concat(" " + lineSpaced[i]);
+            }
+            fullDescription = lineSpaced[1].trim();
+        }
+        CommandType type;
+        switch (lineSpaced[0]) {
+        case "list":
+            type = CommandType.LIST;
+            break;
+        case "add":
+            type = CommandType.ADD;
+            break;
+        case "view":
+            type = CommandType.VIEW;
+            break;
+        case "find":
+            type = CommandType.FIND;
+            break;
+        case "editstep":
+            type = CommandType.EDITSTEP;
+            break;
+        case "editingredient":
+            type = CommandType.EDITINGREDIENT;
+            break;
+        case "delete":
+            type = CommandType.DELETE;
+            break;
+        case "help":
+            type = CommandType.HELP;
+            break;
+        case "clear":
+            type = CommandType.CLEAR;
+            break;
+        case "exit":
+            type = CommandType.EXIT;
+            break;
+        default:
+            type = CommandType.UNKNOWN;
+        }
+        return new Command(type, fullDescription);
+    }
+
+
```

Comment:
Do keep the spacing between methods consistent as well! Some have 1 or 2 lines, some have no lines at all.